src/shared/components/fadeInteraction/FadeInteraction.spec.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from "react";
+import {assert} from 'chai';
+import { mount, shallow } from 'enzyme';
+
+import FadeInteraction from './FadeInteraction';
+
+
+describe('FadeInteraction',function(){
+
+    it('showByDefault prop sets initialShow state properly', function(){
+        const instance = shallow(<FadeInteraction showByDefault={true} />).instance() as FadeInteraction;
+        assert.isTrue(instance.initialShow, 'showByDefault results in initialshow=true');
+        const instance2 = shallow(<FadeInteraction  />).instance() as FadeInteraction;
+        assert.isFalse(instance2.initialShow, 'showByDefault results in initialshow=true');
+    });
+
+    it('#onMouseEnter adjusts state properly', function(){
+        const instance = shallow(<FadeInteraction showByDefault={true} />).instance() as FadeInteraction;
+        assert.isFalse(instance.mouseInside, 'mouseInside is false by default')
+        instance.onMouseEnter();
+        assert.isFalse(instance.initialShow, 'calling onMouseEnter sets initialShow to false');
```

Comment:
same as above, this isn't really testing the functionality. If it's too hard to test visibility directly, I think a better test would be to see if the right class is put on the div, or something like that